server/src/main/java/org/opensearch/index/store/lockmanager/FileBasedMDIndexLockManager.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.store.Lock;
+import org.opensearch.common.Nullable;
+import org.opensearch.index.store.RemoteDirectory;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Lock Manager Class for Index Level Lock
+ * @opensearch.internal
+ */
+public class FileBasedMDIndexLockManager implements RemoteStoreMDIndexLockManager {
```

Comment:
As I think more, can lock manager have static methods for `acquire`, `release`, `isAcquired`? What state is it keeping apart from the `remoteDirectory` instance? Can this be passed as an argument?

Reply:
no.. in that case caller of lock manager needs to create. the lockDirectory instance which is an internal details. considering abstraction, i think we need to have separate lock manager class.